src/Osma.Mobile.App/ViewModels/Connections/ConnectionViewModel.cs

```diff
@@ -50,39 +59,90 @@ public override async Task InitializeAsync(object navigationData)
             await base.InitializeAsync(navigationData);
         }
 
-        public Task RefreshTransactions()
+        public async Task RefreshTransactions()
         {
             RefreshingTransactions = true;
 
-            // TODO OS-201 Get the transactions
-            IList<TransactionItem> transactions = new List<TransactionItem>();
+            var context = await _agentContextProvider.GetContextAsync();
+            var msg = _discoveryService.CreateQuery(context, "*");
 
-            #if DEBUG
-            transactions.Add(new TransactionItem
+            DiscoveryDiscloseMessage protocols = null;
+
+            try
             {
-                Title = "test",
-                Subtitle = "test subtitle",
-                Datetime = "YYYY-MM-DD",
-                Type = TransactionItemType.Actionable.ToString("G")
-            });
-            transactions.Add(new TransactionItem
+                var rsp = await _messageService.SendAsync(context.Wallet, msg, _record, null, true);
+                protocols = rsp.GetMessage<DiscoveryDiscloseMessage>();
+            }
+            catch (Exception) { }
+
+            IList<TransactionItem> transactions = new List<TransactionItem>();
+
+            if (protocols != null)
```

Comment:
Can we invert this check and fail fast to avoid nesting?

Reply:
We will have to duplicate

Reply:
I have reduced the nesting